diff --git a/drivers/gpu/drm/msm/adreno/adreno_device.c b/drivers/gpu/drm/msm/adreno/adreno_device.c
index c3703a51287b46..6107a048ba22fd 100644
--- a/drivers/gpu/drm/msm/adreno/adreno_device.c
+++ b/drivers/gpu/drm/msm/adreno/adreno_device.c
@@ -489,6 +489,7 @@ static const struct adreno_info gpulist[] = {
 			{ 0,   0 },
 			{ 117, 0 },
 			{ 172, 2 }, /* Called speedbin 1 downstream, but let's not break things! */
+			{ 129, 4 },
 			{ 190, 1 },
 		),
 	}, {
