diff --git a/drivers/input/touchscreen/focaltech_touch/focaltech_gesture.c b/drivers/input/touchscreen/focaltech_touch/focaltech_gesture.c
index eaeb4ed447b33d..a679de71eef217 100644
--- a/drivers/input/touchscreen/focaltech_touch/focaltech_gesture.c
+++ b/drivers/input/touchscreen/focaltech_touch/focaltech_gesture.c
@@ -52,7 +52,7 @@
 #define KEY_GESTURE_C                           KEY_C
 #define KEY_GESTURE_Z                           KEY_Z
 #define KEY_GESTURE_CLICK                       KEY_WAKEUP
-#define KEY_GESTURE_FINGER                      KEY_FINGER
+#define KEY_GESTURE_FINGER                      249 // KEY_FINGER //fingerprint
 
 #define GESTURE_LEFT                            0x20
 #define GESTURE_RIGHT                           0x21
