diff --git a/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c b/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c
index 03af04bf25b882..8dc0fa1e782b81 100644
--- a/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c
+++ b/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c
@@ -731,7 +731,7 @@ static ssize_t fts_irq_show(
 {
     ssize_t count = 0;
     struct fts_ts_data *ts_data = dev_get_drvdata(dev);
-    struct irq_desc *desc = irq_to_desc(ts_data->irq);
+    struct irq_desc *desc = irq_data_to_desc(irq_get_irq_data(ts_data->irq));
 
     count = snprintf(buf, PAGE_SIZE, "irq_depth:%d\n", desc->depth);
 
