diff --git a/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c b/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c
index 8dc0fa1e782b81..358dad6e12a64b 100644
--- a/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c
+++ b/drivers/input/touchscreen/focaltech_touch/focaltech_ex_fun.c
@@ -95,7 +95,7 @@ static ssize_t fts_debug_write(
     int writelen = 0;
     int ret = 0;
     char tmp[PROC_BUF_SIZE];
-    struct fts_ts_data *ts_data = PDE_DATA(file_inode(filp));
+    struct fts_ts_data *ts_data = pde_data(file_inode(filp));
     struct ftxxxx_proc *proc = &ts_data->proc;
 
     if (buflen < 1) {
@@ -255,7 +255,7 @@ static ssize_t fts_debug_read(
     int buflen = count;
     u8 *readbuf = NULL;
     u8 tmpbuf[PROC_BUF_SIZE] = { 0 };
-    struct fts_ts_data *ts_data = PDE_DATA(file_inode(filp));
+    struct fts_ts_data *ts_data = pde_data(file_inode(filp));
     struct ftxxxx_proc *proc = &ts_data->proc;
 
     if (buflen <= 0) {
@@ -580,7 +580,7 @@ static int fts_debug_read(
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0))
 static int fts_ta_open(struct inode *inode, struct file *file)
 {
-    struct fts_ts_data *ts_data = PDE_DATA(inode);
+    struct fts_ts_data *ts_data = pde_data(inode);
 
     if (ts_data->touch_analysis_support) {
         FTS_INFO("fts_ta open");
@@ -595,7 +595,7 @@ static int fts_ta_open(struct inode *inode, struct file *file)
 
 static int fts_ta_release(struct inode *inode, struct file *file)
 {
-    struct fts_ts_data *ts_data = PDE_DATA(inode);
+    struct fts_ts_data *ts_data = pde_data(inode);
 
     if (ts_data->touch_analysis_support) {
         FTS_INFO("fts_ta close");
@@ -612,7 +612,7 @@ static ssize_t fts_ta_read(
     struct file *filp, char __user *buff, size_t count, loff_t *ppos)
 {
     int read_num = (int)count;
-    struct fts_ts_data *ts_data = PDE_DATA(file_inode(filp));
+    struct fts_ts_data *ts_data = pde_data(file_inode(filp));
 
     if (!ts_data->touch_analysis_support || !ts_data->ta_buf) {
         FTS_ERROR("touch_analysis is disabled, or ta_buf is NULL");
